### PR TITLE
add configurable logger

### PR DESCRIPTION
**Pull request by diclophis**

### CODE REVIEW

File: lib/unhookd.rb [hunk 1]
```diff
@@ -1,9 +1,4 @@
-require "unhookd/version"
-require "unhookd/configuration"
-require "unhookd/deployer"
-require "unhookd/notifiers/slack"
-require "unhookd/base_values"
-require "unhookd/errors/invalid_configuration"
+#
```
Comment (jasonwc) [inline lib/unhookd.rb]: I'm not familiar with this. Is this convention?

File: lib/unhookd.rb [hunk 2]
```diff
@@ -35,4 +30,12 @@ def self.valid?
   end
 
   private_class_method :valid?
+
+  autoload :Version, "unhookd/version"
```
Comment (jasonwc) [inline lib/unhookd.rb]: 👍

File: lib/unhookd/configuration.rb [hunk 3]
```diff
@@ -15,6 +22,7 @@ def initialize
         title_link: nil,
         message: nil,
       }
+      @logger                = nil # (option) Logger object
```
Comment (jasonwc) [inline lib/unhookd/configuration.rb]: `optional`

File: lib/unhookd/notifiers.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#
+
+module Unhookd
+  module Notifiers
+    autoload :Slack, "unhookd/notifiers/slack"
```
Comment (jasonwc) [inline lib/unhookd/notifiers.rb]: Nice. I'm liking the break out of namespaces here for if we decide to add more in the future.

File: lib/unhookd.rb (hunk 1 above)
Comment (diclophis) [reply lib/unhookd.rb]: its built in? https://ruby-doc.org/core-2.1.0/Module.html#method-i-autoload

we also use it in mavenlink in a couple spots

Comment (jasonwc) [reply lib/unhookd.rb]: I was referring to the `#` at the top of the file haha.

Comment (jasonwc) [reply lib/unhookd.rb]: Autoload is sweet. It makes sense for this use case.

**Pull request merged**